js/index.js

```diff
@@ -1,39 +1,93 @@
 // ITERATION 1
 
+const { ConnectionEmittedEvents } = require("puppeteer");
+
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  
+  const quantity = product.querySelector('.quantity input').innerHTML;
+  const price = product.querySelector('.price span').innerHTML;
+
+  const quantityValue = quantity.innerHTML;
+  const priceValue = price.innerHTML;
+
+  const calcSubtotal = quantityValue * priceValue;
+
+  const displaySubTotal = product.querySelector('.subtotal span');
+
+  displaySubTotal.innerHTML = calcSubtotal;
+
+  return calcSubtotal;
+  
 
-  //... your code goes here
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
+function calculateAll() {
+  
   // ITERATION 2
-  //... your code goes here
+  const productsTotality = document.getElementsByClassName('.product')
+  
+  return updateSubtotal(productsTotality);
+
+ 
+
 
   // ITERATION 3
   //... your code goes here
+
+
+  const totalPrice = document.querySelector(' #total-value span')
+
+  let totalPrice = 0;
+  for(let i = 0; i < productsTotality.length){
+    
+    return 
+  }
+  
+
 }
 
+
+
+
+
+
+
+
+
+
+
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
+  
```

Comment:
En esta parte téniamos que encontrar al objeto padre y eliminarle el hijo:
`function removeProduct(event) {
  const target = event.currentTarget;
  console.log('The target in remove is:', target);
  //... your code goes here
  // node patern sirve para ir una rama hacia arriba en el DOM, lo que hice fue contar las ramas hasta donde queria borrar que era hasta el tr y eran nos ramas por encima del button
  let file = target.parentNode.parentNode
  let parent = file.parentNode
  parent.removeChild(file)
}`